packages/jaeger-ui/src/model/ddg/DdgEVManager.tsx

```diff
@@ -0,0 +1,137 @@
+// Copyright (c) 2019 Uber Technologies, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import { compareKeys } from './visibility-key';
+
+import { PathElem, DdgEdge, DdgVertex, TDdgModel } from './types';
+
+export default class DdgEVManager {
+  private edges: Set<DdgEdge>;
+  private lastVisibilityKey: string;
+  private pathElemToVertex: Map<PathElem, DdgVertex>;
+  private vertices: Map<string, DdgVertex>;
+  private visibilityIdxToPathElem: Map<number, PathElem>;
+
+  constructor({ ddgModel }: { ddgModel: TDdgModel }) {
+    this.edges = new Set();
+    this.lastVisibilityKey = '';
+    this.pathElemToVertex = new Map();
+    this.vertices = new Map();
+    this.visibilityIdxToPathElem = ddgModel.visibilityIdxToPathElem;
+  }
+
+  private addElems = (newIndices: number[]) => {
```

Comment:
Comments on addElems / removeElems would be useful.

Reply:
added